spec/sailor.spec.js

```diff
@@ -176,20 +142,33 @@ describe('Sailor', function () {
                     expect(fakeAMQPConnection.ack.callCount).toEqual(1);
                     expect(fakeAMQPConnection.ack.calls[0].args[0]).toEqual(message);
                 })
-                .done(done, done.fail);
+                .done(done, done); //todo: use done.fail after migration to Jasmine 2.x
         });
 
         it('should send request to API server to update keys', function (done) {
-
+            settings.FUNCTION = 'keys_trigger';
             var sailor = new Sailor(settings);
 
-            spyOn(sailor, "getStepInfo").andReturn({
-                function: "keys_trigger"
-            });
-
-            var nockScope = nock('http://apihost.com:80')
-                .put('/v1/accounts/1234567890', {keys: {oauth: {access_token: 'newAccessToken'}}})
-                .reply(200, "Success");
+            sailor.apiClient = {
```

Comment:
I would agree with you if you would spy on ``tasks.retrieveStep`` and ``accounts.update`` but you are completely replacing it. Just imagine the signature of ``tasks.retrieveStep`` changed after a version upgrade. You would not notice that because you neither do integration testing nor you are spying on the original code.